File: app/src/main/java/io/connorwyatt/flashcards/interfaces/IPerformanceBreakdown.java
package io.connorwyatt.flashcards.interfaces;

public interface IPerformanceBreakdown {
    int getNegativeCount();

    double getNegativePercent();

    int getNeutralCount();

    double getNeutralPercent();

    int getPositiveCount();

    double getPositivePercent();

    int getRatedTotal();

    int getSkipCount();

    double getSkipPercent();

    int getTotal();

    void addOnPerformanceBreakdownChangeListener(OnPerformanceBreakdownChangeListener listener);

    void removeOnPerformanceBreakdownChangeListener(OnPerformanceBreakdownChangeListener listener);

    void clearOnPerformanceBreakdownChangeListener();

    interface OnPerformanceBreakdownChangeListener {
        void onChange();
    }
}

File: app/src/main/java/io/connorwyatt/flashcards/interfaces/IPerformanceBreakdown.kt
package io.connorwyatt.flashcards.interfaces

interface IPerformanceBreakdown
{
    val negativeCount: Int

    val negativePercent: Double

    val neutralCount: Int

    val neutralPercent: Double

    val positiveCount: Int

    val positivePercent: Double

    val ratedTotal: Int

    val skipCount: Int

    val skipPercent: Double

    val total: Int

    fun addOnPerformanceBreakdownChangeListener(listener: OnPerformanceBreakdownChangeListener)

    fun removeOnPerformanceBreakdownChangeListener(listener: OnPerformanceBreakdownChangeListener)

    fun clearOnPerformanceBreakdownChangeListener()

    interface OnPerformanceBreakdownChangeListener
    {
        fun onChange()
    }
}
